Trappist/src/Promact.Trappist.Core/Controllers/AccountController.cs

```diff
@@ -8,11 +13,70 @@ namespace Promact.Trappist.Core.Controllers
 {
     public class AccountController : Controller
     {
-        public IActionResult Login()
+        private readonly IAccountRepository _accountRepository;
+        private readonly IStringConstants _stringConstant;
+
+        public AccountController(IAccountRepository accountRepository, IStringConstants stringConstant)
         {
+            _accountRepository = accountRepository;
+            _stringConstant = stringConstant;
+        }
+
+        /// <summary>
+        /// this method is used to see the view of login
+        /// </summary>
+        /// <returns>Login form view</returns>
+        [HttpGet]
+        [AllowAnonymous]
+        public IActionResult Login(string returnUrl = null)
+        {
+            ViewData["ReturnUrl"] = returnUrl;
             return View();
         }
 
+        /// <summary>
+        ///  This method will be called with  credentials to validate user
+        /// </summary>
+        /// <returns>succesful login-checking for user and redirect to testdashboard page</returns>
+        [HttpPost]
+        [AllowAnonymous]
+        public async Task<IActionResult> Login(Login loginModel, string returnUrl = null)
+        {
+            if (ModelState.IsValid)
+            {
+                if (await _accountRepository.SignIn(loginModel))
+                {
+                    return RedirectToLocal(returnUrl);
+                }
+                else
+                {
+                    ViewBag.Error = _stringConstant.InavalidLoginError;
+                    return View(loginModel);
+                }
+            }
+            else
+            {
+                ModelState.AddModelError(string.Empty, _stringConstant.InavalidModelError);
+                return View(loginModel);
+            }
+        }
+        /// <summary>
+        /// this method is used to rediect to any local url link
```

Comment:
Strictly I don't want any spelling mistakes in comments. Here 'redirect's spelling is wrong.